### PR TITLE
cmake: emu: qemu: introduce `CONFIG_QEMU_GDBSERVER_LISTEN_DEV` so that gdbserver listeners other than `-s`  can be set

### PR DESCRIPTION
**Pull request by alpsayin**
First, we introduce a new string option to pass into QEMU invocation so that gdbserver listeners other than `-s` (tcp::1234) can be set while setting its default to `"tcp::1234"` for backwards compatibility.

Then we remove the enforced `-s` and set `-gdb` but only if above-mentioned Kconfig is NOT an empty string.
This allows users to utilize `QEMU_EXTRA_FLAGS` from shell environment. 
Otherwise, trying to inject a `-gdb` flag from shell causes a fatal clash that occurs between `-s` and `-gdb` during QEMU invocation.

Signed-off-by: Alp Sayin <alpsayin@gmail.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Can you add this into the docs about qemu debug

**Comment by alpsayin** [comment]
It just came to me that this is probably more useful as a Kconfig option. 
If anyone wants to inject at CMake level it'll allow that too.
I'll update the doc too accordingly.

**Review by nordicjm — DISMISSED** [review]

**Comment by stephanosio** [comment]
> Done. But I couldn't find anywhere how to "resolve" this conversation. Is this on you?

@alpsayin Yes, @galak either needs to approve or dismiss his review.

**Review by stephanosio — DISMISSED** [review]

File: boards/Kconfig [hunk 1]
```diff
@@ -99,6 +99,18 @@ config QEMU_UEFI_BOOT
...
+	string "QEMU gdbserver listen device"
+	default "tcp::1234"
+	depends on QEMU_TARGET
+	help
+	  This options is passed onto QEMU as a parameter to `-gdb` option.
+	  The default value is equivalent to `-s` which is a shorthand for
+		`-gdb tcp::1234`. An empty value omits the `-gdb` parameter altogether.
+		This allows the injection of `-gdb` parameter from other sources such
+		as the `QEMU_EXTRA_FLAGS` environment variable. Refer to application
+		development doc and/or QEMU invocation doc for more info.
```
Comment (nordicjm) [inline boards/Kconfig:112]: This is not right, it was correct before, 1 tab followed by 2 spaces

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/Kconfig (hunk 1 above)
Comment (alpsayin) [reply boards/Kconfig]: Wow, you're fast! Yes, I noticed it immediately too (burnt by my editor's consistent tabbing)

**Review by nordicjm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by alpsayin** [comment]
🤦🏼 I'm sorry, I pressed a button and it removed @nashif @tejlmand @carlescufi and @jeremybettis. I blame git for this one.

**Comment by nordicjm** [comment]
> 🤦🏼 I'm sorry, I pressed a button and it removed @nashif @tejlmand @carlescufi and @jeremybettis. I blame git for this one.

Github, full of bugs, never fixed - the microsoft way

**Review by galak — APPROVED** [review]

**Pull request merged**